### ISSUE TITLE
Naturally spawned piglins still contribute to the mob cap when PersistenceRequired is true

### ISSUE THREAD

**Issue by GoldenDelicios** [PaperMC/Paper#4635]
### What behaviour is expected:
Piglins that are persistent (by equipping an item or being named by a name tag) should not contribute to the mob cap. This is the current behavior in vanilla, and for other mobs that can equip items.

### What behaviour is observed:
Piglins appear to be contributing to the mob cap, preventing mobs in the area from spawning.

### Steps/models to reproduce:
In the Nether, spawn enough piglins to exceed the mob cap. Give each piglin an item to equip or a name with a name tag. Leave the area to despawn the surrounding mobs, and return. The piglins will persist, but no other mobs will be able to spawn.

Command blocks I used to reproduce this issue in Creative mode:
```
# Use with a repeating command block. Paper.SpawnReason must be included for the piglin to count as a naturally spawning mob.
summon minecraft:piglin ~ ~ ~ {Paper.SpawnReason:"NATURAL", CanPickUpLoot:1b}`

# Spawns a carved pumpkin for the piglins to equip. May need to run a second time for each to be equipped.
execute at @e[distance=..20] run summon minecraft:item ~ ~ ~ {Item:{id:"minecraft:carved_pumpkin", Count:1b}}
```

### Plugin list:
None

### Paper version:
This server is running Paper version git-Paper-220 (MC: 1.16.3) (Implementing API version 1.16.3-R0.1-SNAPSHOT)

### Anything else:
This issue does not arise with piglins spawned in using /summon without explicitly including {Paper.SpawnReason:"NATURAL"} in the nbt. It also does not arise with other mobs that can equip items, like zombies, skeletons, or pigmen.

**Comment by Hughp135** [comment]
to provide some more feedback as to how this impacts gameplay - this presents issues reducing gold farms that rely on bartering with piglins (as an example). this creates a significant divergence in gameplay from vanilla with the only workaround being to move all piglins to a separate area from the farm.

furthermore, this bug can also be taken advantage of, for example by throwing items to mobs to pick up, effectively spawn proofing a large area just by having the persistent mobs nearby preventing any other mobs from spawning

**Comment by Machine-Maker** [comment]
Fixed in e14aff9b32500efd8dac8bcaa3f0676e01b332fb
